File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {

    //take the first value, set that equal to the low, if it gets lower good
    //take the profit from the low right now (if its higher)

    int low = prices[0];
    int profit = 0;


    for(int day = 0; day < pricesSize; day++){
        if(prices[day] < low){
            low = prices[day];
        }

        if(prices[day] - low > profit){
            profit = prices[day] - low;
        }

    }

    return profit;
    
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        low = prices[0]
        high = 0
        
        for i in range(1, len(prices)):
            if prices[i] < low:
                low = prices[i]
            if prices[i] - low > high:
                high = prices[i] - low
        
        return high
                